### DISCUSSION TITLE
Named parameters support

### DISCUSSION THREAD

**Discussion by bnuzhouwei** [duckdb/duckdb#5388] [General]
Dictionary structure is more suit for entries and objects, but can't pass as position parameters.
Can Add Named parameters support?

i.e:

```python
entry = { "att": 1, "att2": 2 }
obj = sql2obj("select * from table where att=@att and att2=@att2", entry )
```

**Comment by guojs** [comment]
It is very important to my project.

**Reply by SuYao007** [reply]
This is very important! Thanks for my project.

**Reply by jonathanchang1** [reply]
That sounds great!

**Comment by jonathanchang1** [comment]
That sounds great!

**Comment by zhaoqbnu** [comment]
This looks great.
This will help me a lot with my project,thakns

**Comment by hanlinwu** [comment]
I fervently hope the authors consider adding support for dictionary structures in a future release.

**Comment by dhirschfeld** [comment]
Resolved by #5611
